cmake/arch_build.cmake

```diff
@@ -271,22 +271,12 @@ function(prepare)
   if (SIMULATION)
     add_definitions(-DSIMULATION=${SIMULATION})
   endif (SIMULATION)
-  
+
   # Check that PSPNAME, BSPTYPE, and OSTYPE are set properly for this arch
-  if (NOT CFE_SYSTEM_PSPNAME OR NOT OSAL_SYSTEM_OSTYPE)
-    if (CMAKE_CROSSCOMPILING)
-      message(FATAL_ERROR "Cross-compile toolchain ${CMAKE_TOOLCHAIN_FILE} must define CFE_SYSTEM_PSPNAME and OSAL_SYSTEM_OSTYPE")
-    elseif ("${CMAKE_SYSTEM_NAME}" STREQUAL "Linux" OR 
-            "${CMAKE_SYSTEM_NAME}" STREQUAL "CYGWIN")
-      # Export the variables determined here up to the parent scope
```

Comment:
This is important for the `SIMULATION=native` mode to work.

Reply:
Can it be improved such that it only overwrites what isn't supplied?  Seems like overriding all three settings when one isn't supplied isn't very friendly to user configuration.

Reply:
It only sets it if not already set and not crosscompiling.  It shouldn't overwrite anything.

Reply:
Either way, I'll take out.  Seems like it would make sense to just default what isn't defined, vs set all three if one of the two is missing.  Not very friendly/intuitive from my perspective.  If I supply something, I'd expect it to be used.

Reply:
Part of this dates back that in previous CFE builds (before recent merges) only CFE_SYSTEM_PSPNAME and OSAL_SYSTEM_OSTYPE was required.  That's why this "if" statement is only checking that those two are defined.  The OSAL build later confirms that OSAL_SYSTEM_BSPTYPE and OSAL_SYSTEM_OSTYPE are defined (that's the error observed in #629).

It is true that it doesn't handle some but not all being specified but most of the time that's an error anyway, because there is not a default for cross compile.  And for non-cross (native) there is only one right answer anyway which is pc-linux/posix.

I would propose rather than fixing that -- the whole thing could be generally simplified a bit -- we really should only need to specify a PSP type.  We can't mix and match BSP/PSPs and OSs - although an OS can have more than one possible BSP/PSP, the reverse isn't true - each PSP is written for a specific OS.  So by specifying a PSP that in turn can dictate what OS layer to compile, so the toolchain only needs to specify the PSP to use and everything else should fall from that.

But the "default" in arch_build.cmake still needs to be there, to handle the case where _not_ crosscompiling and just building for the host machine, though.  In this case no toolchain file is  used and therefore these variables will be unset (rightfully).

Reply:
I'm all for simplification, and definitely support a default.  It was only the override I had concerns with, and that it looked to need a refresh based on the recent changes.

Reply:
Submitted #632.  Looking at the code this might be fairly trivial/easy to do.